solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -1259,6 +1324,47 @@ contract StandardPoolConverter is ConverterVersion, IConverter, ContractRegistry
         return _amount.mul(_reserveBalance) / _supply;
     }
 
+    /**
+     * @dev returns two samples required for calculating the network fee
+     *
+     * @param currProd  the current product of the balances
+     * @param prevProd  the previous product of the balances
+     *
+     * @return the square root of the current product of the balances
+     * @return the square root of the previous product of the balances
+     */
+    function networkFeePoints(
+        uint256 currProd,
+        uint256 prevProd
+    ) private pure returns (uint256, uint256) {
+        return (
+            currProd > 0 ? MathEx.floorSqrt(currProd) : 0,
+            prevProd > 0 ? MathEx.floorSqrt(prevProd) : 0
+        );
+    }
+
+    /**
+     * @dev returns the network fee ratio
+     *
+     * @param currPoint     the current product of the balances
+     * @param prevPoint     the previous product of the balances
+     * @param networkFee    the relative portion (in PPM units) that
+     * should be taken from the conversion fees accumulated in this pool
+     *
+     * @return numerator of the portion that should be taken from the balances of this pool
+     * @return denominator of the portion that should be taken from the balances of this pool
+     */
+    function networkFeeRatio(
```

Comment:
it makes sense to have a utility function that returns the actual amounts.
I understand we have one case in which we only need one of the reserves, let's discuss that prototype.

Reply:
We have (in their new names, as suggested by you):
A. Function `baseReserveBalance`
B. Function `baseReserveBalances`

Function A is used only in function `addLiquidityReturn`, which takes a single reserve token as input.

So it's going to be difficult to use function B here, since function `addLiquidityReturn` will need to determine what the other reserve token is.

Moreover, we need to revise function `addLiquidityReturn` to take all reserve tokens (and the corresponding amounts) as input, following which - we can get rid of function A altogether.

BTW, I'm not sure how this is related to function `networkFeeRatio` though.
Did you mean to post this comment elsewhere in the code?

Reply:
BTW 2 - combine all of the above with our plan to end support for allowing users to pass input amounts in any order, which means that we will be able to get rid of input reserve tokens in all related functions.

Reply:
ok let's discuss the final option here

Reply:
Removed in https://github.com/bancorprotocol/contracts-solidity/pull/568.

Reply:
Done